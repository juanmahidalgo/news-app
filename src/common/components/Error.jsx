import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Paper, Typography } from '@material-ui/core';
import PropTypes from 'prop-types';

const SVG = ({ className }) =>
  <svg className={className} version="1.1" id="Capa_1" xmlns="http://www.w3.org/2000/svg" xmlnsXlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
    viewBox="0 0 50 50" style={{ enableBackground: 'new 0 0 50 50' }} xmlSpace="preserve">
    <circle style={{ fill: '#D75A4A' }} cx="25" cy="25" r="25" />
    <polyline style={{ fill: 'none', stroke: '#FFFFFF', strokeWidth: 2, strokeLinecap: 'round', strokeMiterlimit: 10 }} points="16,34 25,25 34,16" />
    <polyline style={{ fill: 'none', stroke: '#FFFFFF', strokeWidth: 2, strokeLinecap: 'round', strokeMiterlimit: 10 }} points="16,16 25,25 34,34" />
  </svg>

SVG.propTypes = {
  className: PropTypes.string,
}

const useStyles = makeStyles(theme => ({
  root: {
    padding: theme.spacing(3, 2),
    marginTop: '2rem',
    display: 'flex',
  },
  svg: {
    height: '4rem',
    marginRight: '1rem',
  },
  textWrapper: {
    display: 'flex',
    flexDirection: 'column',
  }
}));

const ErrorContainer = ({ message }) => {
  const classes = useStyles();

  return (
    <Paper className={classes.root}>
      <SVG className={classes.svg} />
      <div className={classes.textWrapper}>
        <Typography variant="h5" component="h3">
          Error
        </Typography>
        <Typography component="p">
          {message}
        </Typography>
      </div>
    </Paper>
  );
}

ErrorContainer.propTypes = {
  message: PropTypes.string,
}

export default ErrorContainer;

