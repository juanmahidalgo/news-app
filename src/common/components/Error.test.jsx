import React from 'react';
import { createRender } from '@material-ui/core/test-utils';
import Error from './Error';

describe('<Error />', () => {
  const render = createRender();

  it('renders correctly', () => {
    const wrapper = render(<Error message={'Error fetching news'} />);
    expect(wrapper).toMatchSnapshot();
  });

});