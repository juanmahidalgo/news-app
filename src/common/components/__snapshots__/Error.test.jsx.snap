// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "MuiPaper-root MuiPaper-elevation1 makeStyles-root-1 MuiPaper-rounded",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "makeStyles-svg-2",
          "id": "Capa_1",
          "space": "preserve",
          "style": "enable-background:new 0 0 50 50",
          "version": "1.1",
          "viewBox": "0 0 50 50",
          "x": "0px",
          "xlink": "http://www.w3.org/1999/xlink",
          "xmlns": "http://www.w3.org/2000/svg",
          "y": "0px",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "cx": "25",
              "cy": "25",
              "r": "25",
              "style": "fill:#D75A4A",
            },
            "children": Array [],
            "name": "circle",
            "namespace": "http://www.w3.org/2000/svg",
            "next": Object {
              "attribs": Object {
                "points": "16,34 25,25 34,16",
                "style": "fill:none;stroke:#FFFFFF;stroke-width:2;stroke-linecap:round;stroke-miterlimit:10",
              },
              "children": Array [],
              "name": "polyline",
              "namespace": "http://www.w3.org/2000/svg",
              "next": Object {
                "attribs": Object {
                  "points": "16,16 25,25 34,34",
                  "style": "fill:none;stroke:#FFFFFF;stroke-width:2;stroke-linecap:round;stroke-miterlimit:10",
                },
                "children": Array [],
                "name": "polyline",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "points": undefined,
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "points": undefined,
                  "style": undefined,
                },
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "points": undefined,
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "points": undefined,
                "style": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "cx": undefined,
              "cy": undefined,
              "r": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "cx": undefined,
              "cy": undefined,
              "r": undefined,
              "style": undefined,
            },
          },
          Object {
            "attribs": Object {
              "points": "16,34 25,25 34,16",
              "style": "fill:none;stroke:#FFFFFF;stroke-width:2;stroke-linecap:round;stroke-miterlimit:10",
            },
            "children": Array [],
            "name": "polyline",
            "namespace": "http://www.w3.org/2000/svg",
            "next": Object {
              "attribs": Object {
                "points": "16,16 25,25 34,34",
                "style": "fill:none;stroke:#FFFFFF;stroke-width:2;stroke-linecap:round;stroke-miterlimit:10",
              },
              "children": Array [],
              "name": "polyline",
              "namespace": "http://www.w3.org/2000/svg",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "points": undefined,
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "points": undefined,
                "style": undefined,
              },
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "cx": "25",
                "cy": "25",
                "r": "25",
                "style": "fill:#D75A4A",
              },
              "children": Array [],
              "name": "circle",
              "namespace": "http://www.w3.org/2000/svg",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "cx": undefined,
                "cy": undefined,
                "r": undefined,
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "cx": undefined,
                "cy": undefined,
                "r": undefined,
                "style": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "points": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "points": undefined,
              "style": undefined,
            },
          },
          Object {
            "attribs": Object {
              "points": "16,16 25,25 34,34",
              "style": "fill:none;stroke:#FFFFFF;stroke-width:2;stroke-linecap:round;stroke-miterlimit:10",
            },
            "children": Array [],
            "name": "polyline",
            "namespace": "http://www.w3.org/2000/svg",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "points": "16,34 25,25 34,16",
                "style": "fill:none;stroke:#FFFFFF;stroke-width:2;stroke-linecap:round;stroke-miterlimit:10",
              },
              "children": Array [],
              "name": "polyline",
              "namespace": "http://www.w3.org/2000/svg",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "cx": "25",
                  "cy": "25",
                  "r": "25",
                  "style": "fill:#D75A4A",
                },
                "children": Array [],
                "name": "circle",
                "namespace": "http://www.w3.org/2000/svg",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "cx": undefined,
                  "cy": undefined,
                  "r": undefined,
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "cx": undefined,
                  "cy": undefined,
                  "r": undefined,
                  "style": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "points": undefined,
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "points": undefined,
                "style": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "points": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "points": undefined,
              "style": undefined,
            },
          },
        ],
        "name": "svg",
        "namespace": "http://www.w3.org/2000/svg",
        "next": Object {
          "attribs": Object {
            "class": "makeStyles-textWrapper-3",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "MuiTypography-root MuiTypography-h5",
              },
              "children": Array [
                Object {
                  "data": "Error",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "h3",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "MuiTypography-root MuiTypography-body1",
                },
                "children": Array [
                  Object {
                    "data": "Error fetching news",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "MuiTypography-root MuiTypography-body1",
              },
              "children": Array [
                Object {
                  "data": "Error fetching news",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "MuiTypography-root MuiTypography-h5",
                },
                "children": Array [
                  Object {
                    "data": "Error",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "h3",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "id": undefined,
          "space": "http://www.w3.org/XML/1998/namespace",
          "style": undefined,
          "version": undefined,
          "viewBox": undefined,
          "x": undefined,
          "xlink": "http://www.w3.org/2000/xmlns/",
          "xmlns": "http://www.w3.org/2000/xmlns/",
          "y": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "id": undefined,
          "space": "xml",
          "style": undefined,
          "version": undefined,
          "viewBox": undefined,
          "x": undefined,
          "xlink": "xmlns",
          "xmlns": "",
          "y": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "makeStyles-textWrapper-3",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "MuiTypography-root MuiTypography-h5",
            },
            "children": Array [
              Object {
                "data": "Error",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "h3",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "MuiTypography-root MuiTypography-body1",
              },
              "children": Array [
                Object {
                  "data": "Error fetching news",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "MuiTypography-root MuiTypography-body1",
            },
            "children": Array [
              Object {
                "data": "Error fetching news",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "p",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "MuiTypography-root MuiTypography-h5",
              },
              "children": Array [
                Object {
                  "data": "Error",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "h3",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "makeStyles-svg-2",
            "id": "Capa_1",
            "space": "preserve",
            "style": "enable-background:new 0 0 50 50",
            "version": "1.1",
            "viewBox": "0 0 50 50",
            "x": "0px",
            "xlink": "http://www.w3.org/1999/xlink",
            "xmlns": "http://www.w3.org/2000/svg",
            "y": "0px",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "cx": "25",
                "cy": "25",
                "r": "25",
                "style": "fill:#D75A4A",
              },
              "children": Array [],
              "name": "circle",
              "namespace": "http://www.w3.org/2000/svg",
              "next": Object {
                "attribs": Object {
                  "points": "16,34 25,25 34,16",
                  "style": "fill:none;stroke:#FFFFFF;stroke-width:2;stroke-linecap:round;stroke-miterlimit:10",
                },
                "children": Array [],
                "name": "polyline",
                "namespace": "http://www.w3.org/2000/svg",
                "next": Object {
                  "attribs": Object {
                    "points": "16,16 25,25 34,34",
                    "style": "fill:none;stroke:#FFFFFF;stroke-width:2;stroke-linecap:round;stroke-miterlimit:10",
                  },
                  "children": Array [],
                  "name": "polyline",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "points": undefined,
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "points": undefined,
                    "style": undefined,
                  },
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "points": undefined,
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "points": undefined,
                  "style": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "cx": undefined,
                "cy": undefined,
                "r": undefined,
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "cx": undefined,
                "cy": undefined,
                "r": undefined,
                "style": undefined,
              },
            },
            Object {
              "attribs": Object {
                "points": "16,34 25,25 34,16",
                "style": "fill:none;stroke:#FFFFFF;stroke-width:2;stroke-linecap:round;stroke-miterlimit:10",
              },
              "children": Array [],
              "name": "polyline",
              "namespace": "http://www.w3.org/2000/svg",
              "next": Object {
                "attribs": Object {
                  "points": "16,16 25,25 34,34",
                  "style": "fill:none;stroke:#FFFFFF;stroke-width:2;stroke-linecap:round;stroke-miterlimit:10",
                },
                "children": Array [],
                "name": "polyline",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "points": undefined,
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "points": undefined,
                  "style": undefined,
                },
              },
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "cx": "25",
                  "cy": "25",
                  "r": "25",
                  "style": "fill:#D75A4A",
                },
                "children": Array [],
                "name": "circle",
                "namespace": "http://www.w3.org/2000/svg",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "cx": undefined,
                  "cy": undefined,
                  "r": undefined,
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "cx": undefined,
                  "cy": undefined,
                  "r": undefined,
                  "style": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "points": undefined,
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "points": undefined,
                "style": undefined,
              },
            },
            Object {
              "attribs": Object {
                "points": "16,16 25,25 34,34",
                "style": "fill:none;stroke:#FFFFFF;stroke-width:2;stroke-linecap:round;stroke-miterlimit:10",
              },
              "children": Array [],
              "name": "polyline",
              "namespace": "http://www.w3.org/2000/svg",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "points": "16,34 25,25 34,16",
                  "style": "fill:none;stroke:#FFFFFF;stroke-width:2;stroke-linecap:round;stroke-miterlimit:10",
                },
                "children": Array [],
                "name": "polyline",
                "namespace": "http://www.w3.org/2000/svg",
                "next": [Circular],
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "cx": "25",
                    "cy": "25",
                    "r": "25",
                    "style": "fill:#D75A4A",
                  },
                  "children": Array [],
                  "name": "circle",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "cx": undefined,
                    "cy": undefined,
                    "r": undefined,
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "cx": undefined,
                    "cy": undefined,
                    "r": undefined,
                    "style": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "points": undefined,
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "points": undefined,
                  "style": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "points": undefined,
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "points": undefined,
                "style": undefined,
              },
            },
          ],
          "name": "svg",
          "namespace": "http://www.w3.org/2000/svg",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "id": undefined,
            "space": "http://www.w3.org/XML/1998/namespace",
            "style": undefined,
            "version": undefined,
            "viewBox": undefined,
            "x": undefined,
            "xlink": "http://www.w3.org/2000/xmlns/",
            "xmlns": "http://www.w3.org/2000/xmlns/",
            "y": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "id": undefined,
            "space": "xml",
            "style": undefined,
            "version": undefined,
            "viewBox": undefined,
            "x": undefined,
            "xlink": "xmlns",
            "xmlns": "",
            "y": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
